snakemake/workflow.py

```diff
@@ -270,11 +271,12 @@ def source_archive(self):
         return self._source_archive
 
     def upload_sources(self):
+        assert self.storage_settings is not None
         with tempfile.NamedTemporaryFile(suffix="snakemake-sources.tar.xz") as f:
             self.write_source_archive(Path(f.name))
             f.flush()
-            with open(f.name, "rb") as f:
-                checksum = hashlib.file_digest(f, "sha256").hexdigest()
+            with open(f.name, "rb") as f1:
```

Comment:
Why this file name `f1`?

Reply:
It is renamed because `f` is used as an `NamedTemporaryFile` object, so I simply renamed the next `f` as it is a different class. I think i should rename the `NamedTemporaryFile` object `f` as `tf` to make it more easier to understand.